sensors/simsensor_common/simsensor_reader.c

```diff
@@ -0,0 +1,297 @@
+	if (err != 0) {
+		return -1;
+	}
+	actField = nextField;
+
+	result->accels.accelZ = reader_getFieldLong(actField, &nextField, &err);
+	if (err != 0) {
+		return -1;
+	}
+	actField = nextField;
+
+	return 0;
+}
+
+
+static int reader_baroDataParse(char *startOfDataSection, time_t timestamp, sensor_event_t *result)
+{
+	char *actField, *nextField;
+	int err = 0;
+
+	actField = startOfDataSection;
+
+
+	result->type = SENSOR_TYPE_BARO;
+	result->timestamp = timestamp;
+
+	result->baro.pressure = reader_getFieldLong(actField, &nextField, &err);
+	if (err != 0) {
+		return -1;
+	}
+	actField = nextField;
+
+	result->baro.temp = reader_getFieldLong(actField, &nextField, &err);
+	if (err != 0) {
+		return -1;
+	}
+	actField = nextField;
+
+	return 0;
+}
+
+
+static time_t reader_nextTimeStampGet(simsens_time_t *timeStruct, time_t timestampFromFile)
+{
+	if (timeStruct->firstTimestamp == READER_TIMESTAMP_NOT_SET) {
+		timeStruct->firstTimestamp = timestampFromFile;
+		timeStruct->lastTimestamp = timestampFromFile;
+	}
+	else if (timeStruct->firstTimestamp == timestampFromFile) {
+		timeStruct->loops++;
+	}
+	else if (timeStruct->lastTimestamp < timestampFromFile) {
+		timeStruct->lastTimestamp = timestampFromFile;
+	}
+
+	return timestampFromFile + timeStruct->loops * (timeStruct->lastTimestamp - timeStruct->firstTimestamp);
+}
+
+
+int reader_read(simsens_reader_t *rd, event_queue_t *queue)
+{
+	char *actField, *nextField;
+	int sensorID, err = 0;
+	size_t curLineLen;
+	time_t timestamp, first_timestamp = READER_TIMESTAMP_NOT_SET;
+	sensor_event_t parsed_result;
+
+	while (eventQueue_full(queue)) {
+		curLineLen = getline(&rd->lineBuf, &rd->bufLen, rd->scenario_file);
+		if (curLineLen == -1) {
+			if (!feof(rd->scenario_file)) {
+				err = -1;
+				break;
+			}
+
+			/* Going to the beginning of the file */
+			if (fseek(rd->scenario_file, 0, SEEK_SET) != 0) {
+				err = -1;
+				break;
+			}
+
+			if (reader_skipHeader(rd) != 0) {
+				err = -1;
+				break;
+			}
+
+			continue;
+		}
+
+		actField = rd->lineBuf;
+
+		sensorID = reader_getFieldLong(actField, &nextField, &err);
+		if (err < 0) {
+			break;
+		}
+
+
+		if ((sensorID & rd->sensor_types) == 0) {
+			if (sensorID == READER_END_SCENARIO_INDICATOR) {
+				break;
+			}
+
+			continue;
+		}
+		actField = nextField;
+
+		timestamp = reader_getFieldLong(actField, &nextField, &err);
+		if (err < 0) {
+			break;
+		}
+		actField = nextField;
+
+		timestamp = reader_nextTimeStampGet(&rd->time_struct, timestamp);
+
+		if (first_timestamp == READER_TIMESTAMP_NOT_SET) {
+			first_timestamp = timestamp;
+		}
+
+		if (timestamp - first_timestamp > rd->time_horizon) {
+			/* Go to start of the line */
+			if (fseek(rd->scenario_file, -curLineLen, SEEK_CUR) != 0) {
+				err = -1;
+			}
+			break;
+		}
+
+		switch (sensorID) {
+			case SENSOR_TYPE_ACCEL:
+				if (reader_accelDataParse(actField, timestamp, &parsed_result) < 0) {
+					err = -1;
+				}
```

Comment:
I think that these parsers offer return code `0/-1` by design. This can be used as `err` setting.